org.adempiere.base/src/org/compiere/model/MArchive.java

```diff
@@ -402,12 +410,26 @@ public File saveAsZip() {
 	 * @param Record_ID
 	 * @param trxName
 	 * @return int[], [0] = report count and [1] = document count
+	 * @deprecated - use {@link #getReportAndDocumentCountByRecordId(int, int, String, String)} instead
 	 */
 	public static int[] getReportAndDocumentCountByRecordId(int AD_Table_ID, int Record_ID, String trxName) {
+		return getReportAndDocumentCountByRecordId(AD_Table_ID, Record_ID, null, trxName);
+	}
+
+	/**
+	 * Get number of document and report archive by table and record id
+	 * 
+	 * @param AD_Table_ID
+	 * @param Record_ID
+	 * @param Record_UU
+	 * @param trxName
+	 * @return int[], [0] = report count and [1] = document count
+	 */
+	public static int[] getReportAndDocumentCountByRecordId(int AD_Table_ID, int Record_ID, String Record_UU, String trxName) {
 		int reportCount = 0;
 		int documentCount = 0;
 		StringBuilder sql = new StringBuilder("SELECT IsReport, COUNT(*) FROM AD_Archive ")
-				.append("WHERE (AD_Table_ID=? AND Record_ID=?) ");
+				.append("WHERE (AD_Table_ID=? AND (Record_ID=? OR Record_UU=?)) ");

```

Comment:
I don't have a big enough data set to analyse this but I'm concern that this would be inefficient not able to make use of the ad_archive_tab_rec index.

Reply:
> I don't have a big enough data set to analyse this but I'm concern that this would be inefficient and not able to make use of the ad_archive_tab_rec index.

Thanks @hengsin 

Improved with commit https://github.com/idempiere/idempiere/pull/1895/commits/cd72987fc0be9cba22714b417cf65a7205a46b49

I inserted 75.000 records in AD_Archive distributed over several tables, records and BPs and tested the query in postgresql.

Effectively, because of the OR clause postgresql was not able to do a proper usage of the new index, as the Record_UU is now basically always filled I think is better to do the query just with that.